forecasttools/data.py

```diff
@@ -0,0 +1,478 @@
+    if os.path.exists(file_save_path):
+        raise FileExistsError(f"The file {file_save_path} already exists.")
+    # create dataset(s) from NHSN data
+    else:
+        # check that a data file exists
+        if not os.path.exists(nhsn_dataset_path):
+            raise FileNotFoundError(
+                f"The file {nhsn_dataset_path} does not exist."
+            )
+        else:
+            # check that the loaded CSV has the needed columns
+            df_cols = pl.scan_csv(nhsn_dataset_path).columns
+            required_cols = [
+                "state",
+                "date",
+                "previous_day_admission_adult_covid_confirmed",
+                "previous_day_admission_influenza_confirmed",
+            ]
+            if len(set(required_cols).intersection(set(df_cols))) != len(
+                required_cols
+            ):
+                raise ValueError(
+                    f"NHSN dataset missing required columns: {set(required_cols) - set(required_cols).intersection(set(df_cols))}"
+                )
+            # fully load and save NHSN dataframe
+            df = pl.read_csv(nhsn_dataset_path)
+            # change date formatting to ISO8601
+            df = df.with_columns(df["date"].str.replace_all("/", "-"))
+            # pathogen specific df saving
+            if dataset == "COVID":
+                df_covid = (
+                    df.select(
+                        [
+                            "state",
+                            "date",
+                            "previous_day_admission_adult_covid_confirmed",
+                        ]
+                    )
+                    .rename(
+                        {
+                            "previous_day_admission_adult_covid_confirmed": "hosp"
+                        }
+                    )
+                    .sort(["state", "date"])
+                )
+                df_covid.write_csv(file_save_path)
+            if dataset == "flu":
+                df_flu = (
+                    df.select(
+                        [
+                            "state",
+                            "date",
+                            "previous_day_admission_influenza_confirmed",
+                        ]
+                    )
+                    .rename(
+                        {"previous_day_admission_influenza_confirmed": "hosp"}
+                    )
+                    .sort(["state", "date"])
+                )
+                df_flu.write_csv(file_save_path)
+            print(
+                f"The file {file_save_name} has been created at {save_directory}."
+            )
+
+
+def make_nhsn_fitted_forecast_idata(
+    nhsn_dataset_path: str,
+    save_directory: str,
+    file_save_name: str,
+    start_date: str = "2022/08/08",
+    end_date: str = "2023/12/08",
+    forecast_days: int = 28,
+    juris_subset: list[str] = ["TX"],
+    create_save_directory: bool = False,
```

Comment:
probably cleaner to have one argument specifying the full path to the save file

Reply:
Thinking it over; I agree, fewer checks, one less parameter.